tests/SprykerEcoTest/Zed/PunchoutCatalogs/Business/PunchoutCatalogsFacadeTest.php

```diff
@@ -73,6 +79,65 @@ public function testFindConnectionByIdRetrievesConnectionWhenItExists(): void
         $this->assertEquals($idPunchoutCatalogConnection, $punchoutCatalogConnectionTransfer->getIdPunchoutCatalogConnection());
     }
 
+    /**
+     * @return void
+     */
+    public function testFindConnectionByIdRetrievesPasswordFormVaultWhenItExists(): void
+    {
+        // Arrange
+        $idPunchoutCatalogConnection = $this->tester->havePunchoutCatalogConnection([
+            PunchoutCatalogConnectionTransfer::NAME => static::CONNECTION_NAME,
+            PunchoutCatalogConnectionTransfer::USERNAME => static::CONNECTION_USERNAME,
+            PunchoutCatalogConnectionTransfer::TYPE => static::CONNECTION_TYPE,
+            PunchoutCatalogConnectionTransfer::FORMAT => static::CONNECTION_FORMAT,
+            PunchoutCatalogConnectionTransfer::PASSWORD => static::CONNECTION_PASSWORD,
+            PunchoutCatalogConnectionTransfer::FK_COMPANY_BUSINESS_UNIT => $this->tester->createCompanyBusinessUnit()
+                ->getIdCompanyBusinessUnit(),
+        ])->getIdPunchoutCatalogConnection();
+
+        // Act
+        $punchoutCatalogConnectionTransfer = $this->tester->getFacade()
+            ->findConnectionById($idPunchoutCatalogConnection);
+
+        // Assert
+        $this->assertNotNull($punchoutCatalogConnectionTransfer);
+        $this->assertEquals(static::CONNECTION_PASSWORD, $punchoutCatalogConnectionTransfer->getPassword());
+    }
+
+    /**
+     * @return void
+     */
+    public function testCreateConnectionCreatesStoresPasswordToVault(): void
+    {
+        // Arrange
+        $companyBusinessUnitEntity = $this->tester->createCompanyBusinessUnit();
+        $punchoutCatalogConnectionTransfer = (new PunchoutCatalogConnectionTransfer())
+            ->setFkCompanyBusinessUnit($companyBusinessUnitEntity->getIdCompanyBusinessUnit())
+            ->setName(static::CONNECTION_NAME)
+            ->setUsername(static::CONNECTION_USERNAME)
+            ->setPassword(static::CONNECTION_PASSWORD)
+            ->setType(static::CONNECTION_TYPE)
+            ->setFormat(static::CONNECTION_FORMAT);
+
+        // Act
+        $punchoutCatalogResponseTransfer = $this->tester->getFacade()
+            ->createConnection($punchoutCatalogConnectionTransfer);
+
+        // Assert
+        $this->assertTrue($punchoutCatalogResponseTransfer->getIsSuccessful());
+
+        $password = $this->tester->getLocator()
```

Comment:
This is not a part of `assert`. Please re-arrange the test / move this logic to another test / move it to the separate method, and then have something like:
```
    // Assert
    $this->assertTrue($punchoutCatalogResponseTransfer->getIsSuccessful());
    $this->assertEquals(static::CONNECTION_PASSWORD, $this->retrievePasswordFromVault(<YOUR_VARIABLE>));
```